### DISCUSSION TITLE
New Integration: AdoraDish V4000

### DISCUSSION THREAD

**Discussion by hrsl01** [siku2/hass-vzug#7] [Device Support]
First of all many thanks for the integration - great job!

I have successfully integrated AdoraDry V2000 and AdoraWash V2000. The data looks good so far but I will check it over the next days to make sure they work properly. Afterwards, these models can be added to the supported device list.

However, the integration of my AdoraDish V4000 fails during the setup. What information can I provide to double check the difference to the supported V6000 to eventually add it to the integration?

Thanks.

**Comment by siku2** [comment]
Can you start by enabling debug logging for the integration and then posting them here? I made sure that the debug logs contain as much information as possible.

**Comment by phoebos02** [comment]
Update on this. My V4000 works.

**Reply by siku2** [reply]
Can you tell me what appliance it is? Do you mean the AdoraDish V4000?

**Reply by phoebos02** [reply]
Yes.

**Comment by hrsl01** [comment]
> Can you start by enabling debug logging for the integration and then posting them here? I made sure that the debug logs contain as much information as possible.

See attached the log when I integrate the V4000. Thanks for your review.
[log_V4000.txt](https://github.com/siku2/hass-vzug/files/12417699/log_V4000.txt)

**Reply by siku2** [reply]
Cool thanks, I found the issue that is causing this. Will release a new version with a fix in a bit.

**Reply by siku2** [reply]
Please let me know if the new version addresses the issue.

**Reply by hrsl01** [reply]
Thanks for the fast feedback and update. I updated my installation, however it still does not work. It results in a different error though. See the log attached.
[log_V4000_2.txt](https://github.com/siku2/hass-vzug/files/12423300/log_V4000_2.txt)

**Reply by siku2** [reply]
Interesting, that's still the same underlying issue. The integration is trying to read the available command categories of the appliance, but it keeps getting an empty list back.
I've noticed that one of my appliances does this at times too, which is why the response is getting rejected and the integration tries again.
It gives up after the 5th failed attempt though, as you can see.

There's always the option of disabling that sanity check, but first I would like to figure out why exactly it doesn't appear to have any categories.

Does your device have the page <http://192.168.2.50/Benutzereinstellungen.html>? If yes, is it just empty?

If you're comfortable with sharing this, a recording of the network traffic when you visit this page would be very useful so I can see which endpoints the frontend uses.
Here's a guide from Zendesk on how to do exactly that: <https://support.zendesk.com/hc/en-us/articles/4408828867098-Generating-a-HAR-file-for-troubleshooting>. (side note: after opening the network monitor you should refresh the page so the requests are actually getting logged)
Unless you've set a password for your appliance (which would break the integration anyway), the HAR file shouldn't contain any sensitive information.

**Reply by hrsl01** [reply]
I checked the webpage - the configuration is filled normally. I did the network record and sent it to you via mail. Hope thats ok with you.

**Reply by siku2** [reply]
@daphil012, thanks I received it. This seems to confirm that the categories are just straight up always empty... 

For comparison, this is what it looks like for my AdoraWash V6000:
![image](https://github.com/siku2/hass-vzug/assets/16530283/7185879e-734a-473d-a357-3e9933af19f0)

Based on what I'm seeing in your recording, your page only ever contains the "Datum & Zeiteinstellungen" section. Is that correct?

**Reply by hrsl01** [reply]
Ah yes, I don't have 6000 version, so therefore I did no know that there are more configurations available. It looks like this:
![image](https://github.com/siku2/hass-vzug/assets/81329842/06bd3e5e-f220-4eb5-abff-e0f179e839fc)

**Reply by siku2** [reply]
@daphil012 released a new version which should hopefully allow your device to be set up properly.

**Reply by hrsl01** [reply]
@siku2 Great, many thanks for the update. Integration was succesful. I will check the data over the next days but it looks promising. Thank you.
